File: Data Structures & Algorithms/coin-change-ii/submission-3.cpp
class Solution {
public:
    int change(int amount, vector<int>& coins) {
        vector<vector<int>> dp(coins.size()+1, vector<int>(amount+1, 0));
        int n = coins.size();
        sort(coins.begin(), coins.end());
        for(int i = 0; i <= coins.size(); i++){
                dp[i][0] = 1;
        }

        for(int i = n-1; i >= 0; i--){
            for(int j = 0; j<=amount; j++){
                if(coins[i]<=j){
                dp[i][j]=dp[i+1][j];
                dp[i][j]+=dp[i][j-coins[i]];
                }
            }
        }
        return dp[0][amount];
    }
};

File: Data Structures & Algorithms/meeting-schedule-ii/submission-3.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def minMeetingRooms(self, intervals: List[Interval]) -> int:
        intervals.sort(key = lambda i : i.start)
        room = 0
        minheap = []
        for i in range(len(intervals)):
            if minheap and minheap[0] <= intervals[i].start:
                heapq.heappop(minheap)
            heapq.heappush(minheap, intervals[i].end)
        return len(minheap)
